src/components/Editor.jsx

```diff
@@ -137,6 +163,64 @@ class Editor extends React.Component {
     this._editor.moveCursorTo(0, 0);
     this._resizeEditor();
   }
+
+  _autoIndent(event) {
+    if (!this.Beautify) {
+      return;
+    }
+
+    const {session, selection} = this._editor;
+    const doc = session.getDocument();
+    const range = selection.getRange();
+
+    const options = {};
+    if (session.getUseSoftTabs()) {
+      options.indent_char = ' ';
+      options.indent_size = session.getTabSize();
+    } else {
+      options.indent_char = '\t';
+      options.indent_size = 1;
+    }
+
+    const startIndex = doc.positionToIndex(range.start);
+    const endIndex = doc.positionToIndex(range.end);
+
+    const source = session.getValue();
+
+    const {
+      code: newSource,
+      startIndex: newStartIndex,
+      endIndex: newEndIndex,
+    } = format(
+      this.Beautify,
+      source,
+      startIndex,
+      endIndex,
+      this.props.language,
+      options,
+    );
+
+    // Make sure we set the new value for the editor via editor.setValue
+    // instead of session.setValue otherwise the undo/redo stack will be
+    // clobbered!
+    this._editor.setValue(newSource);
+    const newRange = Range.fromPoints(
+      doc.indexToPosition(newStartIndex),
+      doc.indexToPosition(newEndIndex),
+    );
+
+    selection.setSelectionRange(newRange);
+
+    event.preventDefault();
+  }
+
+  _handleKeyPress(event) {
+    if (this.Beautify &&
+      event.key === 'i' && (event.metaKey || event.ctrlKey) &&
```

Comment:
I think `|| event.ctrlKey` is redundant since we also check `&& !event.ctrlKey` on the next line